domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1609,79 +1609,113 @@ public TotaledResults(ResultSet set, Integer total) {
    * @param replyHandler
    */
   @SuppressWarnings({"unchecked", "squid:S00107"})
-  public <T> void streamGet(String table, T entity, String fieldName, CQLWrapper filter, boolean returnIdField,
-      String distinctOn, Handler<T> streamHandler, Handler<AsyncResult<Void>> replyHandler) {
+  public <T> void streamGet(String table, T entity, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    Handler<T> streamHandler, Handler<AsyncResult<Void>> replyHandler) {
+
+    streamGet(table, entity, fieldName, filter, returnIdField, distinctOn,
+      res -> {
+        if (res.failed()) {
+          replyHandler.handle(Future.failedFuture(res.cause()));
+          return;
+        }
+        PostgresClientStreamResult streamResult = res.result();
+        streamResult.handler(streamHandler);
+        streamResult.endHandler(x -> replyHandler.handle(Future.succeededFuture()));
+      });
+  }
+
+  public <T> void streamGet(String table, T entity, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    Handler<AsyncResult<PostgresClientStreamResult>> replyHandler) {
+
+    streamGet(table, entity, fieldName, filter, returnIdField, distinctOn,
+      Collections.emptyList(), replyHandler);
+  }
+
+  @SuppressWarnings({"unchecked", "squid:S00107"})
+  public <T> void streamGet(String table, T entity, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    List<FacetField> facets, Handler<AsyncResult<PostgresClientStreamResult>> replyHandler) {
 
     client.getConnection(res -> {
       if (res.succeeded()) {
         SQLConnection connection = res.result();
         Class<T> clazz = (Class<T>) entity.getClass();
-        doStreamGet(connection, false, table, clazz, fieldName, filter, returnIdField, distinctOn,
-            Collections.emptyList(), streamHandler, replyHandler);
+        doStreamGet(connection, false, table, clazz, fieldName, filter, returnIdField,
+          distinctOn, facets, replyHandler);
       } else {
         replyHandler.handle(Future.failedFuture(res.cause()));
       }
     });
   }
 
+  private void closeIfNotTransaction(SQLConnection connection, boolean transaction) {
+    if (!transaction) {
+      connection.close();
+    }
+  }
+
   @SuppressWarnings({"unchecked", "squid:S00107"})
-  private <T> void doStreamGet(SQLConnection connection, boolean transactionMode, String table, Class<T> clazz,
-      String fieldName, CQLWrapper wrapper, boolean returnIdField, String distinctOn, List<FacetField> facets,
-      Handler<T> streamHandler, Handler<AsyncResult<Void>> replyHandler) {
+  private <T> void doStreamGet(SQLConnection connection, boolean transactionMode,
```

Comment:
Can you avoid using transactionMode? We have closeAndHandleResult. The caller of a method should be in charge for both opening and closing the connection. Closing the connection in the method is error-prone because there are several places and one can easily miss one.